libs/beacon/src/beacon_complaints_manager.cpp

```diff
@@ -0,0 +1,319 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "beacon/beacon_complaints_manager.hpp"
+
+#include <cstdint>
+#include <mutex>
+
+namespace fetch {
+namespace beacon {
+
+constexpr char const *LOGGING_NAME = "DKGComplaints";
+
+void ComplaintsManager::ResetCabinet(MuddleAddress const &address, uint32_t threshold)
+{
+  FETCH_LOCK(mutex_);
```

Comment:
We should at some point review which of these classes we actually need to be thread safe, probably this doesn't need to be the way it's used